site/src/pages/WorkspacesPage/WorkspacesTable.tsx

```diff
@@ -55,13 +61,46 @@ export const WorkspacesTable: FC<WorkspacesTableProps> = ({
 }) => {
 	const theme = useTheme();
 	const dashboard = useDashboard();
+	const workspaceIDToAppByStatus = useMemo(() => {
+		return (
+			workspaces?.reduce(
+				(acc, workspace) => {
+					if (!workspace.latest_app_status) {
+						return acc;
+					}
+					for (const resource of workspace.latest_build.resources) {
+						for (const agent of resource.agents ?? []) {
+							for (const app of agent.apps ?? []) {
+								if (app.id === workspace.latest_app_status.app_id) {
+									acc[workspace.id] = { app, agent };
+									break;
+								}
+							}
+						}
+					}
+					return acc;
+				},
+				{} as Record<
+					string,
+					{
+						app: WorkspaceApp;
+						agent: WorkspaceAgent;
+					}
+				>,
+			) || {}
+		);
+	}, [workspaces]);
+	const hasAppStatus = useMemo(
+		() => Object.keys(workspaceIDToAppByStatus).length > 0,
+		[workspaceIDToAppByStatus],
+	);
```

Comment:
Should this be hidden behind a feature flag?

Reply:
This is only visible if there are reported statuses, which only occur if you use the exp command anyways. Otherwise, the dashboard remains 100% the same as it was before.